lib/chef/resource/aws_cache_cluster.rb

```diff
@@ -0,0 +1,31 @@
+require 'chef/provisioning/aws_driver/aws_resource'
+
+class Chef::Resource::AwsCacheCluster < Chef::Provisioning::AWSDriver::AWSResource
+  aws_sdk_type AWS::ElastiCache
+
+  attribute :cluster_name, kind_of: String, name_attribute: true
+  attribute :az_mode, kind_of: String
+  attribute :preferred_availability_zone, kind_of: String
+  attribute :preferred_availability_zones,
```

Comment:
Why do we need both `:prefered_availability_zone` and `:prefered_availability_zones` if the plural version can also take in a String?  Seems like its just more work to have to do a diff later to see which the user provided and whether one has priority over the other or should we thrown an error if both are provided, etc.